src/main/java/org/pesmypetcare/webservice/dao/medalmanager/UserMedalDao.java

```diff
@@ -0,0 +1,58 @@
+package org.pesmypetcare.webservice.dao.medalmanager;
+
+import org.pesmypetcare.webservice.entity.medalmanager.UserMedalEntity;
+import org.pesmypetcare.webservice.error.DatabaseAccessException;
+import org.pesmypetcare.webservice.error.DocumentException;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Oriol Catalán
+ */
+public interface UserMedalDao {
+
+    /**
+     * Gets a medal identified by its name and owner.
+     * @param owner Username of the owner of the medal
+     * @param name Name of the medal
+     * @return The MedalEntity corresponding to the owner's medal data
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     * @throws DocumentException When the document does not exist
+     */
+    UserMedalEntity getUserMedalData(String owner, String name) throws DatabaseAccessException, DocumentException;
+
+    /**
+     * Gets the data from all the specified medals from the database.
+     * @param owner Username of the owner of the medals
+     * @return The List containing all the owner medals data
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     * @throws DocumentException When the document does not exist
+     */
+    List<Map<String, UserMedalEntity>> getAllUserMedalsData(String owner) throws DatabaseAccessException, DocumentException;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 124).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5517612)